various improvements

mprenditore opened the pull request:
## Description

Thank you for the work on this role. I've started to use it and I've found some improvements that everyone I'm sure will appreciate.
I didn't know how to run the automated tests, I've tried to run `molecule` but it wasn't finding the ANSIBLE_LIBRARY even if ansible is installed and working, please let me know how can I check it.

- added tags support: `setup`, `upd_conf`, `unsecure_logs`
- added possibility to show registration logs via `unsecure_logs` tag. See PR #11 
- moved *global* to *global_values* in ***gitlab_runner_config***
- added *global_strings* to ***gitlab_runner_config***
- changed the task *Set advanced configuration*  to loop through values
- changed *extra_options*  to array and loop through it
- updated README

I'm open to discussion about how to improve what I've implemented. For now it's working on my system but there is always a way to optimize.

Cheers

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Reviews


- @vutkin 
- @tgadiev 
- @kharkevich 

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass with my changes

mprenditore committed:
various improvements

- added tags support: `setup`, `upd_conf`, `unsecure_logs`
- added possibility to show registration logs via `unsecure_logs` tag
- moved *global* to *global_values* in ***gitlab_runner_config***
- added *global_strings* to ***gitlab_runner_config***
- changed the task *Set advanced configuration*  to loop through values
- changed *extra_options*  to array and loop through it
- updated README

pavelpikta commented on README.md:
```diff
@@ -27,46 +27,56 @@ Requirements
 
 With [playlabs](https://yourlabs.io/oss/playlabs) you can install this role with just one command, ie:
 
-    playlabs install lean_delivery.gitlab_runner @localhost gitlab_ci_token=yourcommand gitlab_host=yourlabs.io gitlab_runner_limit=4 gitlab_version=11.6 
```
Should be reverted empty space

mprenditore replied:
> Should be reverted empty space

The empty space at the end of the lines? What's the purpose?

pavelpikta replied:
README formatting ![image](https://user-images.githubusercontent.com/9392033/66826545-593ded00-ef55-11e9-917b-023b306997cd.png)

mprenditore replied:
got it, I've setup my editor to trim automatically the whitespace. I don't think I'm the only one that does it, what about overriding the whitespaces with a `<br>`?

pavelpikta replied:
> got it, I've setup my editor to trim automatically the whitespace. I don't think I'm the only one that does it, what about overriding the whitespaces with a `<br>`?

You could add README.md to ignore automatically trim spaces

mprenditore replied:
> > got it, I've setup my editor to trim automatically the whitespace. I don't think I'm the only one that does it, what about overriding the whitespaces with a `<br>`?
> 
> You could add README.md to ignore automatically trim spaces

Thanks didn't know how to do it. Done and I'm going to push the changes

mprenditore replied:
done

pavelpikta commented on tasks/config-gitlab-runner.yml:
```diff
@@ -1,37 +1,62 @@
+
+- name: enabling unsecure logging
+  set_fact:
+    no_logs: false
```
I think we could create a new variable for manage logging and don't use tags

mprenditore replied:
> I think we could create a new variable for manage logging and don't use tags

I think tags are a better choice instead of variables because the main purpose is act differently specifying a single value in the cmd and I think are even easier to remember.

vutkin replied:
Is using the vars be more elegant way to get same result and should take less extra code?
```
ansible-playbook playbook.yml --extra-vars "no_logs=true"  --tags ...
```

vutkin replied:
Anyway this approach should be documented in readme.

mprenditore replied:
Both approaches are fine for me, I used the tag because I see the variable approach to be more suitable for overriding the values at inventory level, instead to change the behaviour of the playbook, the tag should be more correct. The amount of code added I don't think cause any problem.
I can add it to the documentation once it's decided the approach.

mprenditore replied:
@pavelpikta @vutkin  how should we implement it?
Let's decide and I'll make the changes.

tgadiev replied:
I think, using vars is more flexible, since you can define them in various ways: inventory, playbook, extra-vars etc.

mprenditore replied:
done, updated also the documentation adding the variable

pavelpikta commented on tasks/config-gitlab-runner.yml:
```diff
@@ -1,37 +1,62 @@
-      value: '{{ global_section.concurrent }}'
-  vars:
-    global_section: '{{ gitlab_runner_config.global }}'
+  with_dict: "{{ gitlab_runner_config.global_values | default({}) }}"
```
Why with_dict instead of loop?

mprenditore replied:
We could use `loop` but in that case we've to add the filter *dict2items* at the end, but I think using *with_dict* is more self explanatory.

pavelpikta replied:
> We could use `loop` but in that case we've to add the filter _dict2items_ at the end, but I think using _with_dict_ is more self explanatory.

Ansible recommend use loop for most use cases.

mprenditore replied:
> > We could use `loop` but in that case we've to add the filter _dict2items_ at the end, but I think using _with_dict_ is more self explanatory.
> 
> Ansible recommend use loop for most use cases.

My bad, I didn't read the suggestion of moving to *loop* from *with_X*.
Going to implement it

pavelpikta commented on tasks/config-gitlab-runner.yml:
```diff
@@ -1,37 +1,62 @@
+    line: '{{ item.key }} = "{{ item.value }}"'
+    state: present
+  notify: restart gitlab-runner
+  with_dict: "{{ gitlab_runner_config.global_strings | default({}) }}"
```
Why with_dict instead of loop?

mprenditore replied:
We could use `loop` but in that case we've to add the filter *dict2items* at the end, but I think using *with_dict* is more self explanatory.

pavelpikta replied:
Ansible recommend use loop for most use cases.

mprenditore replied:
My bad, I didn't read the suggestion of moving to *loop* from *with_X*.
Going to implement it.

mprenditore replied:
done

mprenditore committed:
added back the trailing spaces in the list

mprenditore committed:
moved *with_dict* to *loop*

mprenditore committed:
changed behavor for not showing sensitive info

To be able to show sensitive information while executing the playbook instead of the tag `unsecure_logs` just specify the variable `no_logs` to false

tgadiev commented:
Merged. Thanks @mprenditore !

mprenditore commented:
> Merged. Thanks @mprenditore !

Awesome, welcome!
If I'll add new features in the future I'mm make other MRs.

Cheers!
